File: leetcode/easy/206/206-iterative.js
/**
 * Runtime: 88 ms, faster than 32.32%
 * Memory Usage: 40.7 MB, less than 37.74%
 */

/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  let prev = null;
  let curr = head;

  while(curr != null) {
    let temp = curr.next;
    curr.next = prev;
    prev = curr;
    curr = temp;
  }

  return prev;
};

File: leetcode/easy/206/206-iterative.rb
# Runtime: 96 ms, faster than 7.00%
# Memory Usage: 210.4 MB, less than 37.74%

# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def reverse_list(head)
  prev = nil
  curr = head

  while(curr != nil) do
    temp = curr.next
    curr.next = prev
    prev = curr
    curr = temp
  end

  return prev
end
